core/templates/services/keyboard-shortcut.service.ts

```diff
@@ -67,13 +67,13 @@ export class KeyboardShortcutService {
     });
 
     Mousetrap.bind('c', function() {
-      document.getElementById('categoryBar').focus();
-      return false;
+      var categoryBar = <HTMLElement>document.querySelector(
+        '.protractor-test-search-bar-dropdown-toggle');
+      categoryBar.focus();
     });
 
     Mousetrap.bind('s', function() {
       document.getElementById('skipToMainContentId').focus();
```

Comment:
> > Why we need setHref method in keyboard shortcut service?
>
> We need to wrap window.location.href in order to successfully mock the function for the frontend test.
https://stackoverflow.com/questions/4792281/mocking-window-location-href-in-javascript

Should we use window-ref.service.ts in test/spec file?

Reply:
I don't think we need to use it in the spec file, because I can already spyOn the setHref function. However, with a recommendation from @sagangwee I switched to using 'window-ref.service.ts'  in the 'keyboard-shortcuts.service' file just now.

Reply:
Could you check if I am importing windows-ref.service.ts properly? I couldn't follow the convention used in 'core/templates/services/contextual/url.service.ts' because it would make navigation shortcuts not work at all.

Thanks!

Reply:
Can you do similar to this file: core/templates/domain/sidebar/sidebar-status.service.spec.ts

Reply:
is it resolved?

Reply:
Sorry!I didn't notice this comment. Hmm, Isn't that the same way as url.service.ts?